app/views/works/_work_header_navigation.html.erb

```diff
@@ -51,10 +51,15 @@
       <%= link_to ts("Cancel Bookmark"), "#", :class => "hidden bookmark_form_placement_close" %>
     </li>
 
-    <% unless current_user.is_author_of?(@work) || current_user.try(:preference).try(:history_enabled) == false || marked_for_later?(@work) %>
-      <li class="mark"><%= marktoread_link(@work) %></li>
+    <% unless current_user.is_author_of?(@work) || current_user.try(:preference).try(:history_enabled) == false %>
+     <li class="mark"> 
+      <% if marked_for_later?(@work) %>
```

Comment:
Also, FWIW, this code will fix the bug with pressing the button multiple times causing a toggle effect, but I believe it will NOT fix the bug with the button appearing wrongly, because I believe that happens because Reading objects are written to the db in batches on a delay, and therefore "marked_for_later?" returns false even after the object has been marked for later.